assets/js/modules/analytics/settings/settings-edit.js

```diff
@@ -113,13 +114,15 @@ export default function SettingsEdit() {
 		};
 	}, [] );
 
+	const usingProxy = directSelect( 'core/site' ).isUsingProxy();
```

Comment:
This should be using `useSelect`. We should never call `Data.select` directly from within a component.

Reply:
Since this never changes, can I use select outside the component? If I use `useSelect` what would the dependency be?

Reply:
We shouldn't assume that data from the datastore will never change. Even if we don't explicitly change it via an action, all selectors will return `undefined` until resolved which is still a change to the return value that the component needs to be subscribed to. In most cases with simple data like this, the data will be preloaded so the resolution should be instant, but we shouldn't assume that. It's okay to use `useSelect` without any dependencies as this is different than an empty array which means "only run once".

Reply:
Ok, got it! thanks for clarifying, I will adjust these and remove the use of the direct `select` approach.

Reply:
Removed all direct `select` usage in favor of `useSelect`